### PR TITLE
Add typestubs for python-periphery

### PR DESCRIPTION
**Pull request by AussieSeaweed**
I added complete type stubs for the [python-periphery](https://pypi.org/project/python-periphery/) package.

### CODE REVIEW

File: stubs/python-periphery/periphery/gpio.pyi [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        bias: str = ...,
+        drive: str = ...,
+        inverted: bool = ...,
+        label: str | None = ...,
+    ) -> None: ...
+    def __new__(self, path: str, line: int | str, direction: str, **kwargs: Any) -> CdevGPIO: ...  # noqa: Y034
+
+class SysfsGPIO(GPIO):
+    GPIO_OPEN_RETRIES: int
+    GPIO_OPEN_DELAY: float
```
Comment (srittau) [inline stubs/python-periphery/periphery/gpio.pyi:61]:
```suggestion
    GPIO_OPEN_RETRIES: ClassVar[int]
    GPIO_OPEN_DELAY: ClassVar[float]
```

(Needs import from `typing`.)

File: stubs/python-periphery/periphery/mmio.pyi [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from ctypes import c_void_p
+from types import TracebackType
+
+long = int
+
+class MMIOError(IOError): ...
+
+class MMIO:
+    def __init__(self, physaddr: int | long, size: int | long, path: str = ...) -> None: ...
```
Comment (srittau) [inline stubs/python-periphery/periphery/mmio.pyi:9]: Just change `int | long` to `int` (here and below) as it's basically just an alias for `int | int`.

File: stubs/python-periphery/periphery/pwm.pyi [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from types import TracebackType
+
+class PWMError(IOError): ...
+
+class PWM:
+    PWM_STAT_RETRIES: int
+    PWM_STAT_DELAY: float
```
Comment (srittau) [inline stubs/python-periphery/periphery/pwm.pyi:7]:
```suggestion
    PWM_STAT_RETRIES: ClassVar[int]
    PWM_STAT_DELAY: ClassVar[float]
```

(needs import)

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, a few small things below.

Did you consider submitting this directly to python-periphery? The maintainer seems open to adding type hints, see vsergeev/python-periphery#47. Cc @vsergeev

**Comment by vsergeev** [comment]
> Thanks, a few small things below.
> 
> Did you consider submitting this directly to python-periphery? The maintainer seems open to adding type hints, see python-periphery#46. Cc @vsergeev

Would happily take it in python-periphery, has been on the todo list and I'm working on a 2.4.0 release now.

**Comment by AussieSeaweed** [comment]
> > Thanks, a few small things below.
> > Did you consider submitting this directly to python-periphery? The maintainer seems open to adding type hints, see python-periphery#46. Cc @vsergeev
> 
> Would happily take it in python-periphery, has been on the todo list and I'm working on a 2.4.0 release now.

I created a PR here: vsergeev/python-periphery#59. I will close this.

**Pull request closed without merging**